### PR TITLE
workflow: l4lb: pass correct path for PR checkout

### PR DESCRIPTION
**Pull request by jibi**
Passing ${{ github.workspace }}/pull-request to the init.sh script as
path of the PR checkout is incorrect, as inside the VM the repository
will be under a different path.

Instead of that, pass the correct absolute path for the PR checkout.

Moreover, checkout first the upstream repo (under cilium/cilium) and
only after that the PR one (under cilium/cilium/pull-request), as
otherwise the upstream checkout will nuke the directory with PR one.

Fixes: #19953

### CODE REVIEW

**Review by nbusseneau — APPROVED** [review]
I would suggest we pass in the whole path (`../../pull-request/install/kubernetes/cilium`) as a proper `HELM_CHART_DIR` arg to the script, to help make the script and execution context independents, but this works as-is.

**Review by tklauser — APPROVED** [review]

**Comment by jibi** [comment]
CI is green: https://github.com/cilium/cilium/runs/6655411820?check_suite_focus=true

related changes:
v1.10: https://github.com/cilium/cilium/pull/20008
v1.11: https://github.com/cilium/cilium/pull/20009

File: test/l4lb/test.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@ set -eux
 
 IMG_OWNER=${1:-cilium}
 IMG_TAG=${2:-latest}
-PULL_REQUEST_CHECKOUT=${3:-../..}
+HELM_CHART_DIR=${3:-/vagrant/pull-request/install/kubernetes/cilium}
```
Comment (nbusseneau) [inline test/l4lb/test.sh]: For backward compatibility: 
```suggestion
HELM_CHART_DIR=${3:-/vagrant/install/kubernetes/cilium}
```

File: test/nat46x64/test.sh [hunk 2]
```diff
@@ -4,7 +4,7 @@ set -eux
 
 IMG_OWNER=${1:-cilium}
 IMG_TAG=${2:-latest}
-PULL_REQUEST_CHECKOUT=${3:-../..}
+HELM_CHART_DIR=${3:-/vagrant/pull-request/install/kubernetes/cilium}
```
Comment (nbusseneau) [inline test/nat46x64/test.sh]: Likewise.

**Review by nbusseneau — APPROVED** [review]
LGTM, one nit.

**Pull request merged**